### PR TITLE
Update the release documentation

### PR DESCRIPTION
**Pull request by anfredette**
I've updated this documentation based on notes from the past two releases.  We should try to follow the steps described in this document for the next release and take another pass at updating it if necessary.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -64,177 +72,303 @@ A release for the bpfman project is comprised of the following major components:
...
+After these steps are completed, the following should occur:
+- All [GitHub actions][gh-actions] should complete successfully.
+- The release appears on the GitHub Releases Page.
+- Images are built and updated with the new version tag at:
+    - quay.io/bpfman
+    - quay.io/bpfman-bytecode
+    - quay.io/bpfman-userspace
+- The new version appears at [crates.io](https://crates.io/crates/bpfman)
+- A new RPM is built and pushed to the [bpfman COPR
+  repository](https://copr.fedorainfracloud.org/coprs/g/ebpf-sig/bpfman/).
```
Comment (msherif1234) [inline RELEASE.md:199]: we also we need to undo the changes in bpfman/src/lib.rs and pointed back to latest right ? or revert the PR

File: RELEASE.md [hunk 2]
```diff
@@ -64,177 +72,303 @@ A release for the bpfman project is comprised of the following major components:
...
-Additionally you can navigate to the github release page and draft a release
-generating release notes there as well.
 
-### Release Steps
+After the release is complete do the following:
 
-The following steps must be done by one of the [bpfman maintainers][bpfman-team],
-please always try and cut a release candidate before pushing an official major
-or minor release.
+- Run `make build-release-yamls` and add the yaml files generated to the release
```
Comment (msherif1234) [inline RELEASE.md:204]: pls make it clear u will run the above make command from `examples` directory

**Review by msherif1234 — COMMENTED** [review]
did u intentionally remove the update to the community operator ?

File: RELEASE.md (hunk 2 above)
Comment (anfredette) [reply RELEASE.md]: Done.

**Comment by anfredette** [comment]
> did u intentionally remove the update to the community operator ?

I think it's in there.  See line 261 of new doc.  If not, let me know what to add, or we can add the instructions in an update after we do the next release.

> also we have now split repo so should we split the release doc or create common docs repo under and have that doc there as we as any general doc ?

Good point.  We can discuss, but maybe we add a RELEASE.md file to bpfman-operator that says something like, we synchronize the releases.  Do bpfman first.  And, point to the instructions in bpfman?

File: RELEASE.md [hunk 3]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
+Releases are tagged in git with the version number prefixed by "v".  For
+example, release version 0.5.2 is tagged as v0.5.2.
+
+### Kubernetes API Versions (e.g. v1alpha2, v1beta1)
+
+Within the bpfman-operator, API versions are primarily used to indicate the
+stability of a resource. For example, if a resource has not yet graduated to
+beta, it is still possible that it could either be removed from the API or
+changed in backward incompatible ways. For more information on API versions,
+refer to the [full Kubernetes API versioning
```
Comment (Billy99) [inline RELEASE.md:100]:
```suggestion
refer to the full [Kubernetes API versioning
```
I would remove "full" from the link

File: RELEASE.md [hunk 4]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
-weight process meaning many of the versioned manifests do not necessarily need
-to be created.
-
-- Open an update PR that:
-    - Adds a new changelog for the release
-    - Updates the [Cargo.toml](https://github.com/bpfman/bpfman/blob/main/Cargo.toml) version for the workspace:
-        - `version = "x.x.x"`
-        - `bpfman = { version = "x.x.x", path = "./bpfman" }"`
-        - `bpfman-api = { version = "x.x.x", path = "./bpfman-api" }`
+To simplify the generation of the release notes details, we recommend using the
```
Comment (Billy99) [inline RELEASE.md:139]:
```suggestion
To simplify the generation of the release notes details, we are using the
```

File: RELEASE.md [hunk 5]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
-    - Runs `cargo generate-lockfile`
-    - Updates the bpfman-operator version in it's [Makefile](https://github.com/bpfman/bpfman-operator/blob/main/Makefile):
-        - `VERSION ?= x.x.x`
-- Make sure CI is green and merge the update PR.
-- Create a tag using the `HEAD` of the `main` branch. This can be done using the `git` CLI or
-  Github's [release][release] page.
-- Tag the release using the commit on `main` where the changelog update merged.
-  This can  be done using the `git` CLI or Github's [release][release]
+    - Run `cargo generate-lockfile`
+    - Add a new `examples` config directory for the release version.
```
Comment (Billy99) [inline RELEASE.md:175]:
```suggestion
    - Add a new `bpfman/examples/config/v0.x.y/` and `bpfman/examples/config/v0.x.y-selinux/` config directory for the release version by copying the latest release directory and running a search for the current release and replace with the new release.```

File: RELEASE.md [hunk 6]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
+This section describes the standard release process used when making a release
+from the main branch and may be used for major, minor, or patch releases.  As
+mentioned above, we first complete the release for `bpfman` and then follow that
+up with a release for `bpfman-operator`.
+
+#### bpfman Release
+
+- Create a new branch in your `bpfman` fork named, for example
+  `<githubuser>/release-x.y.z`, and use the new branch in the upcoming steps.
+- Make the following changes
```
Comment (Billy99) [inline RELEASE.md:164]: Question:
Do you think it is worth adding a step to search though the code for the current version, `0.5.1` (without the `v`) and replace with the current release. There is documentation that references releases like this file, loading from RPM, etc, that may be nice to always point to latest release in the instructions? There are a few spots pointing to 0.4.2 currently, so if we add this step, we should do a one time scrub for spots with older releases too.

File: RELEASE.md [hunk 7]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
+      - `go-uprobe-counter-install-selinux.yaml`
+      - `go-uretprobe-counter-install-selinux.yaml`
+      - `go-kprobe-counter-install-selinux.yaml`
+      - `go-app-counter-install-selinux.yaml`
+- Do another PR that changes the tag for `XDP_DISPATCHER_IMAGE` and
+`TC_DISPATCHER_IMAGE` in `bpfman/src/lib.rs` back to `latest`.
+
+#### bpfman-operator Release
+
+- Create a new branch in your `bpfman` fork named, for example
```
Comment (Billy99) [inline RELEASE.md:229]:
```suggestion
- Create a new branch in your `bpfman-operator` fork, for example
```

File: RELEASE.md [hunk 8]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
+### Standard Release from Main Branch
+
+This section describes the standard release process used when making a release
+from the main branch and may be used for major, minor, or patch releases.  As
+mentioned above, we first complete the release for `bpfman` and then follow that
+up with a release for `bpfman-operator`.
+
+#### bpfman Release
+
+- Create a new branch in your `bpfman` fork named, for example
```
Comment (Billy99) [inline RELEASE.md:162]:
```suggestion
- Create a new branch in your `bpfman` fork, for example
```

File: RELEASE.md [hunk 9]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
         - Note: `bpfman-csi` does not need to be updated.
-    - Runs `cargo generate-lockfile`
-    - Updates the bpfman-operator version in it's [Makefile](https://github.com/bpfman/bpfman-operator/blob/main/Makefile):
-        - `VERSION ?= x.x.x`
-- Make sure CI is green and merge the update PR.
-- Create a tag using the `HEAD` of the `main` branch. This can be done using the `git` CLI or
-  Github's [release][release] page.
-- Tag the release using the commit on `main` where the changelog update merged.
-  This can  be done using the `git` CLI or Github's [release][release]
+    - Run `cargo generate-lockfile`
```
Comment (Billy99) [inline RELEASE.md:174]: - Update the bpfman version in the [examples/Makefile][examples-makefile]:
    - `VERSION ?= x.y.z`

And add link at bottom of file.

File: RELEASE.md (hunk 6 above)
Comment (anfredette) [reply RELEASE.md:164]: That makes sense.

Comment (anfredette) [reply RELEASE.md:164]: Actually, this is a little tricky.  There are a lot of places where the old release needs to remain, and there are other dependencies that just happen to have the same version number as the current or older bpfman releases.

Comment (Billy99) [reply RELEASE.md:164]: So the one time scrub might a bit tricky maybe. Is there anywhere that needs to stay at 0.5.1? But then going forward, just change current release to next release everywhere

File: RELEASE.md [hunk 10]
```diff
@@ -64,177 +72,304 @@ A release for the bpfman project is comprised of the following major components:
...
-### Overview
+### Release Version Number
+
+`bpfman` uses the MAJOR.MINOR.PATCH scheme for version numbers in which the
+components are defined as follows:
 
-Each new release of bpfman is defined with a "bundle version" that
-represents the Git tag of a release, such as `v0.4.2`. This contains the
-components described above
+MAJOR: Incremented for incompatible API changes.
```
[the remark this answers is not in the document]
Comment (anfredette) [reply RELEASE.md]: Done

File: RELEASE.md (hunk 3 above)
Comment (anfredette) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 4 above)
Comment (anfredette) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 8 above)
Comment (anfredette) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 9 above)
Comment (anfredette) [reply RELEASE.md:176]: Done.

File: RELEASE.md (hunk 5 above)
Comment (anfredette) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 7 above)
Comment (anfredette) [reply RELEASE.md]: Done.

File: RELEASE.md (hunk 6 above)
Comment (anfredette) [reply RELEASE.md:166]: I added a bullet, but whoever is using it will need to use their judgment.  You may find the string in the following places that shouldn't be changed: changelogs, cargo.toml, cargo.lock, Containerfile.bpfman.openshift (which has a different version), example config files (at least)

**Comment by anfredette** [comment]
Thanks for your reviews @msherif1234  and @Billy99!  I think I've addressed all of your comments.  Let me know if I missed anything.

**Review by Billy99 — APPROVED** [review]
